gui/src/app/state/psbt.rs

```diff
@@ -167,23 +168,15 @@ impl PsbtState {
                 return cmd;
             }
             Message::View(view::Message::Spend(view::SpendTxMessage::Broadcast)) => {
-                let outpoints: HashSet<_> = self.tx.coins.keys().cloned().collect();
+                let outpoints: Vec<_> = self.tx.coins.keys().cloned().collect();
                 return Command::perform(
                     async move {
                         daemon
-                            // TODO: filter for the outpoints in `tx.coins` when this is possible:
-                            // https://github.com/wizardsardine/liana/issues/677
-                            .list_coins()
+                            .list_coins(&[CoinStatus::Spending], &outpoints)
                             .map(|res| {
                                 res.coins
                                     .iter()
-                                    .filter_map(|c| {
-                                        if outpoints.contains(&c.outpoint) {
-                                            c.spend_info.map(|info| info.txid)
-                                        } else {
-                                            None
-                                        }
-                                    })
+                                    .map(|c| c.spend_info.expect("coin is spending").txid)
```

Comment:
I would prefer we keep a filter_map and no expect

Reply:
I've changed this to filter_map.